### PR TITLE
[DependencyInjection] Introduce "list" env var processor

### PR DESCRIPTION
**Pull request by amenophis**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | symfony/symfony-docs#18402

This MR allows to take an arbitrary list of values seperated by comma and parse it as an array with each item typed.
The implementation is just a hack that wrap the list with hooks and parse it as json

```
parameters:
    env(VALUES): 1,2.0,3.1,1e1
    values: '%env(list:VALUES)%' # array(4) { [0]=> int(1) [1]=> float(2) [2]=> float(3.1) [3]=> float(10) }
```

### CODE REVIEW

**Comment by ro0NL** [comment]
IIUC this creates incompatible types:

https://github.com/symfony/symfony/blob/e1cc5f294c1029106824a0ad763fe3e2e6020d3c/src/Symfony/Component/DependencyInjection/EnvVarProcessor.php#L46-L47

of which im not sure what the side effects are :sweat_smile:

**Comment by nicolas-grekas** [comment]
I agree with @ro0NL, this would break expectations. Maybe with a new `int_array`? Or just filter on your side? You have to anyway since php arrays aren't typed...

**Comment by ro0NL** [comment]
> since php arrays aren't typed

with psalm they are :)

this is why i prefer `json:`, to control the types by value, eg. `[1, "1", true]`

maybe for `csv:` auto casting some string values to php types (like `XmlUtils` does) is reasonable?

**Comment by ro0NL** [comment]
another idea is to deprecate `csv:` in favor of new `list:` which could forward to json wrapped in brackets :)

**Comment by amenophis** [comment]
@ro0NL Thanks for the suggestion, this way is more clean.
@nicolas-grekas I can deprecate the csv prefix, WDYT ?

**Comment by nicolas-grekas** [comment]
Please don't deprecate the CSV processor, it fits a use case.

**Comment by ro0NL** [comment]
i forgot to mention, we could detect unquoted strings in list and quote them before passing to json

but let's keep them separate :+1:

**Comment by amenophis** [comment]
PR Doc added, both can be merged if OK for you @ro0NL and @nicolas-grekas :)

**Comment by nicolas-grekas** [comment]
Looking at the code, I'm sorry but I don't see the point of adding this processor. Just use json if you have the choice of the format you put into your env var.

**Comment by amenophis** [comment]
@nicolas-grekas thanks for the feedback, closing the PR

**Comment by welcoMattic** [comment]
IIUC, this new `list` processor will wrap the list of values between `[]` to use json_decode. What if in the value, I have something like `foo[bar]`? Should I refer to the JSON syntax and adding `"` around this value? If yes, I agree with @nicolas-grekas, we could use `json` processor. I think it will be difficult (impossible?) to avoid to refer to the JSON syntax for values with special characters.

To assert that, could you add some tests with values containing `[` or `{` with and without wrapping around `"`?

**Comment by ro0NL** [comment]
@welcoMattic `list: 1,2,3` and `json: [1,2,3]` would be equivalent, thus follow same error handling

my main drive here was to call a list a list :sweat_smile:, and compatibility with existing envs, while bringing json capabilities

TLDR: just use json :) but having `list: 1,"2",true,string,"also string"` would be noble :angel:
